controller/src/beerocks/master/db/node.h

```diff
@@ -211,6 +212,29 @@ class node {
     std::unordered_map<sMacAddr, std::unordered_map<sMacAddr, link_metrics_data>>
         m_link_metric_data;
 
+    class ap_metrics_data {
+    public:
+        ap_metrics_data(){};
+        ~ap_metrics_data(){};
+
+        sMacAddr bssid;
+        uint8_t channel_utilization;
+        uint16_t number_of_stas_currently_associated;
+        uint8_t estimated_service_info_field_ac_be[3];
+        uint8_t estimated_service_info_field_ac_bk[3];
+        uint8_t estimated_service_info_field_ac_vo[3];
+        uint8_t estimated_service_info_field_ac_vi[3];
+
+        bool add_ap_metric_data(std::shared_ptr<wfa_map::tlvApMetric> ApMetricData);
```

Comment:
I don't really like that the wfa_map definitions leak through into the database. I would prefer if the add_ap_metric_data function was in the code that deals with wfa_map stuff - in other words, in the master. Admittedly, that file is already way too large, but we can split it up.

Note that this is just my opinion based on a very vague feeling of appropriateness, so you shouldn't necessarily take it into account.